src/resources/js/download.js

```diff
@@ -27,6 +28,32 @@ $.get("/api/packages").done(function(json) {
 			
 			var $optpkg = $(optpkgTemplate);
 			$('.optpkg-name', $optpkg).text(pkg.path);
+			if (preselectedPackage.includes(pkg.path)) {
+				$('.optpkg-check', $optpkg)[0].checked = true;
+				$('.optpkg-check', $optpkg).closest('.optpkg').toggleClass('selected');
+			}
+
+			$('.optpkg-check', $optpkg).change({pkg: pkg}, (event) => {
+				const element = $(event.currentTarget);
+				let currentSelected = getAllPackageFromURLSearchDecode();
+				const pkgPath = event.data.pkg.path;
+				if (element.is(':checked')) {
+					if (!currentSelected.includes(pkgPath)) {
+						currentSelected = [...currentSelected, pkgPath];
+					}
+				} else {
+					if (currentSelected.includes(pkgPath)) {
+						const position = currentSelected.indexOf(pkgPath);
+						currentSelected.splice(position, 1);
+					}
+				}
+				const encodeArray = encodeArrayURI(currentSelected);
+				let newUrl = window.location.pathname;
+				if (encodeArray && encodeArray.length > 0) {
+					newUrl = `${window.location.pathname}?package=${encodeArray.join('&package=')}`
```

Comment:
You should use `(new URLSearchParams(window.location.search)).append("package", val)` instead to generate a URL such as `download?package=github.com/foo/bar&package=github.com/foo/baz`.

Reply:
Done, Thanks ! :)